src/state/farms/elasticv2/hooks.ts

```diff
@@ -0,0 +1,258 @@
+import { NativeCurrencies } from 'constants/tokens'
+import { useActiveWeb3React } from 'hooks'
+import { useContract, useMulticallContract } from 'hooks/useContract'
+import { useKyberswapConfig } from 'hooks/useKyberswapConfig'
+import { useAppDispatch, useAppSelector } from 'state/hooks'
+import { isAddressString } from 'utils'
+
+import { defaultChainData, setFarms, setLoading, setUserFarmInfo } from '.'
+import { ElasticFarmV2, SubgraphFarmV2, SubgraphToken, UserFarmV2Info } from './types'
+
+const queryFarms = gql`
+  {
+    farmV2S {
+      id
+      startTime
+      endTime
+      pool {
+        id
+        feeTier
+        tick
+        sqrtPrice
+        liquidity
+        reinvestL
+        token0 {
+          id
+          symbol
+          name
+          decimals
+        }
+        token1 {
+          id
+          symbol
+          name
+          decimals
+        }
+      }
+      rewards(orderBy: index, orderDirection: asc) {
+        token {
+          id
+          symbol
+          name
+          decimals
+        }
+        amount
+        index
+      }
+      ranges {
+        id
+        index
+        isRemoved
+        tickUpper
+        tickLower
+        weight
+      }
+    }
+  }
+`
+
+let isSubscribed = false
+
+const positionManagerInterface = new Interface(NFTPositionManagerABI.abi)
+
+export const useElasticFarmsV2 = (subscribe = false) => {
+  const dispatch = useAppDispatch()
+  const { networkInfo, isEVM, chainId, account } = useActiveWeb3React()
+  const elasticFarm = useAppSelector(state => (chainId ? state.elasticFarmV2[chainId] : defaultChainData))
+  const { elasticClient } = useKyberswapConfig(chainId)
+
+  const multicallContract = useMulticallContract()
+  const farmv2QuoterContract = useContract(
+    isEVM ? (networkInfo as EVMNetworkInfo).elastic.farmv2Quoter : undefined,
+    FarmV2QuoterABI,
+  )
+
+  const [getElasticFarmV2, { data, error }] = useLazyQuery(queryFarms, {
+    client: elasticClient,
+    fetchPolicy: 'network-only',
+  })
+
+  useEffect(() => {
+    if (isEVM && !elasticFarm.farms && !elasticFarm.loading) {
+      dispatch(setLoading({ chainId, loading: true }))
+      getElasticFarmV2().finally(() => {
+        dispatch(setLoading({ chainId, loading: false }))
+      })
+    }
+  }, [isEVM, chainId, dispatch, getElasticFarmV2, elasticFarm])
+
+  useEffect(() => {
+    const i =
+      subscribe && !isSubscribed
+        ? setInterval(() => {
+            isSubscribed = true
+            getElasticFarmV2()
+          }, 20_000)
+        : undefined
+    return () => {
+      i && clearInterval(i)
+      isSubscribed = false
+    }
+  }, [subscribe, dispatch, getElasticFarmV2])
+
+  useEffect(() => {
+    if (error && chainId) {
+      dispatch(setFarms({ chainId, farms: [] }))
+      dispatch(setLoading({ chainId, loading: false }))
+    }
+  }, [error, dispatch, chainId])
+
+  useEffect(() => {
+    if (data?.farmV2S && chainId) {
```

Comment:
no need to check chainId
```suggestion
    if (data?.farmV2S) {
```